ggshield/core/config.py

```diff
@@ -156,29 +156,36 @@ def save(self, config_path: str) -> None:
         self.save_yaml(config_path)
 
     @classmethod
-    def load(cls, config_path: Optional[str] = None) -> "UserConfig":
+    def load(cls, config_path: Optional[str] = None) -> Tuple["UserConfig", str]:
         """
         Load the various user configs files to create a UserConfig object:
         - global user configuration file (in the home)
         - local user configuration file (in the repository)
+
+        Returns a UserConfig instance, and the path where updates should be saved
         """
 
         user_config = UserConfig()
-        # Load the user config
         if config_path:
             data = load_yaml(config_path) or {}
             user_config.update_config(data)
-        else:
-            for global_config_filename in GLOBAL_CONFIG_FILENAMES:
-                global_config_path = get_global_path(global_config_filename)
-                global_data = load_yaml(global_config_path)
-                if global_data and user_config.update_config(global_data):
-                    break
-            for local_config_path in LOCAL_CONFIG_PATHS:
-                local_data = load_yaml(local_config_path)
-                if local_data and user_config.update_config(local_data):
-                    break
-        return user_config
+            return user_config, config_path
+
+        for global_config_filename in GLOBAL_CONFIG_FILENAMES:
+            global_config_path = get_global_path(global_config_filename)
+            global_data = load_yaml(global_config_path)
+            if global_data and user_config.update_config(global_data):
+                break
+
+        for local_config_path in LOCAL_CONFIG_PATHS:
+            local_data = load_yaml(local_config_path)
+            if local_data and user_config.update_config(local_data):
+                config_path = local_config_path
+                break
+
+        if config_path is None:
+            config_path = DEFAULT_LOCAL_CONFIG_PATH
+        return user_config, config_path
```

Comment:
isn't it a breaking change to load global+local after the specified config file? the specified config file should stop the others from loading.

Reply:
No because in the case of the specified config file there is an early return (line 176). But your comment makes me think it's probably not-obvious, I can remove the return and use an `else` block like the original code.

Reply:
Yes i looked for it quickly but didn't find it. Not used to review on GitHub.

I'm ok with it do as you wish.